GoogleForCreators/web-stories-wp
# PPC V 2.0: Expand Panels When Checkpoint is Hit

## Context

Part of new prepublish checklist requirements. 

## Summary

This is a design request left over from the 1.5 version of the checklist that we had to cut because the old checklist existed in the inspector panel and overriding visibility of tabs there was something that went against the rest of the app because it's set in local storage by user preference. Now that we're not tied by that it makes way more sense to expand new panels for the users in the checklist since otherwise nothing would be open. 

Panels only update this way when a new checkpoint is hit - so only twice total in the span of a story. 


## Relevant Technical Choices

- when checkpoint updates in useCheckpoint set corresponding panel to be expanded.


## To-do

Nothing :) 

## User-facing changes

None unless the new checklist is enabled. 

| Toggle between states | Kept Open | Straight to Publish |
| --- | --- | --- |
| https://user-images.githubusercontent.com/10720454/124200971-90d32c80-da8b-11eb-936f-6ff2b23027c5.mp4 | https://user-images.githubusercontent.com/10720454/124200972-90d32c80-da8b-11eb-8d78-089bf0ee7c00.mp4 | https://user-images.githubusercontent.com/10720454/124200973-916bc300-da8b-11eb-9dd6-a5c497d9ec05.mp4 |



## Testing Instructions

Try triggering checkpoints in a story and see that when the checkpoint is hit the corresponding panel expands. So, when you add a second page or any other element the design panel (and a11y if you have anything that would trigger it - by default on blank pages nothing shows up for accessibility) show up and the design panel is expanded. When you add a 5th page or try to publish a story and click 'review checklist' instead you get the priority panel and it's expanded. 

Try closing that panel and opening another one. Close the checklist, reopen it, your selection should stay intact. 

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #8132
